lib/lightning/oban_manager.ex

```diff
@@ -0,0 +1,203 @@
+  #             exit_code: 7,
+  #             log:
+  #               ~s"""
+  #               #{message}:
+  #               #{log}
+  #               Exited with #{code}
+  #               """
+  #               |> String.split("\n")
+  #           }
+
+  #         _other ->
+  #           %Engine.Result{
+  #             exit_code: 5,
+  #             log: [
+  #               "OpenFn encountered an unexpected error during the execution of this job:",
+  #               inspect(error, pretty: true)
+  #             ]
+  #           }
+  #       end
+  #     end
+
+  #   result
+  #   |> RunService.ensure_result_handled(run)
+  #   |> FlowService.ensure_flow_controlled(state)
+  #   |> clean_timeouts_for_run()
+  # end
+
+  # @doc """
+  # Attempt to create a record of the crash, updating an existing run in the db.
+  # """
+
+  # def handle_orphaned_run(run_id, state) do
+  #   run = Repo.get(Run, run_id)
+
+  #   %Engine.Result{
+  #     exit_code: 4,
+  #     log: [
+  #       "==== LOST CONNECTION TO NODE ===================================================",
+  #       "",
+  #       "We lost connection to the node that was handling this run. It may have taken",
+  #       "longer than the grace period. Here's what to do:",
+  #       "",
+  #       " - Check your destination system to ensure it's working and responding properly",
+  #       "   to API requests.",
+  #       " - Check your job expression to make sure you haven't created any infinite loops",
+  #       "   or long sleep/wait commands.",
+  #       "",
+  #       "Only enterprise plans support runs lasting more than 100 seconds.",
+  #       "Contact enterprise@openfn.org to enable long-running jobs."
+  #     ]
+  #   }
+  #   |> RunService.ensure_result_handled(run)
+  #   |> FlowService.ensure_flow_controlled(state)
+  # end
+
+  # @doc """
+  # Given a run which we are certain has been handled properly (i.e., exit_code
+  # has been set, FlowService has been called), clean_timeouts_for_run will remove
+  # the 'discarded' Oban jobs related to that run from the DB. Normally, discarded
+  # Oban Jobs stay in the DB to be manually removed during daily reviews, but exit
+  # code 4 (an unresponsive NodeVM) happens frequently enough that we should
+  # automate this process. A record of unresponsive VMs can (and should) be
+  # reviewed to ensure that it happens infrequently, but this decouples that
+  # process from the normal operation of the platform which, sadly, includes and
+  # handles unresponsive NodeVMs as part of its standard operating procedure.
+  # """
+
+  # def clean_timeouts_for_run(%Run{id: id}) do
+  #   Logger.warn(
+  #     "Deleting discarded ObanJob(s) for Run ##{id}; it failed with Oban.TimeoutError and was handled."
+  #   )
+
+  #   ObanQuery.timeouts_for_run(id)
+  #   |> Repo.delete_all()
+  # end
+
+  defp scrub_context(context) do
+    {_private_state, safe_context} = pop_in(context, [:args, "state"])
+    safe_context
+  end
+end
```

Comment:
I have no idea what this is for in the context of Lightning, we don't pass state around in our job arguments. Unless theres something else we use it for please remove it.